docs/enhancements/add-feature-support-by-generic-attributes.md

```diff
@@ -0,0 +1,144 @@
+of that specific feature.
+- Change the current API to not return whole list but to return the support-level of a specific feature filtered by 
+some attributes (e.g. openshift_version, cpu_architecture, and more)
+- Get per cluster feature-support level - by given only cluster ID the service will determine the support level by 
+the cluster already exists attributes without the need to set multiple properties
+
+Implementation example:
+
+### swagger.yaml
+```swagger codegen
+  /v2/support-level/{feature_id}:
+  parameters:
+    - in: path
+      name: feature_id
+      required: true
+    - in: query
+      name: cluster_id
+    - in: query
+      name: openshift_version
+    - in: query
+      name: cpu_architecture
+  responses:
+    "200":
+      description: Success.
+      schema:
+        $ref: '#/definitions/support_level'
+
+  support_level:
+    type: string
+    enum: [supported, unsupported, tech-preview, dev-preview]
+```
+
+- This option will allow us to get the feature support level per feature with cluster or specific properties (openshift_version or cpu_architecture) 
+sending both cluster_id and one of the other parameters will result in a bad request.
+
+### features.go
+```go
+func GetFeatureSupportLevel(openshiftVersion string, featureId models.FeatureSupportLevel, architecture string) models.SupportLevel {
+	return FeaturesList[featureId].GetSupportLevel(openshiftVersion, architecture)
+}
+```
+
+
+### features.go
+```go
+type SupportLevelFeature interface {
+  GetId() models.FeatureSupportLevel
+  GetSupportLevel(openshiftVersion, architecture string) models.SupportLevel
+}
+
+type NutanixIntegrationFeature struct{}
+
+func (feature *NutanixIntegrationFeature) GetId() models.FeatureSupportLevel {
+	return models.FeatureSupportLevelNUTANIXINTEGRATION
+}
+
+func (feature *NutanixIntegrationFeature) GetSupportLevel(openshiftVersion, architecture string) models.SupportLevel {
+	if isNotSupported, err := common.BaseVersionLessThan("4.11", openshiftVersion); isNotSupported || err != nil {
+		return models.SupportLevelUnsupported
+	}
+
+	if isEqual, _ := common.BaseVersionEqual("4.11", openshiftVersion); isEqual {
+		return models.SupportLevelDevPreview
+	}
+	return models.SupportLevelSupported
+}
+```
+
+
+### User Stories
+
+#### Story 1
+
+As a developer I want to be able to easily filter the feature-support level by feature id and multiple attributes.
+
+### Risks and Mitigations
+
+As with every large code and API change, this change contains a risk of breaking existing functionality.
+This can be mitigated by adding unit and subsystem tests to various scenarios.
+
+
+### UI Impact
+
+UI will to take impact from this change.\
+Currently, UI get the feature-support level list and save it locally, and for every feature they query that "db" 
+for each feature.\
+This change will force the UI to change the way they handle the feature-support level (due to the API change),
```

Comment:
Not sure if it's mentioned anywhere, but if we have additional API endpoints we need to mark the old one deprecated and remove it when we move to v3 (just like we did on the move from v1->v2)

Reply:
If we break an API, our policy is to provide backward compatibility for 3 months.  We wouldn't move to v3 unless we dramatically changed models.

Reply:
Of course. It's just that deprecation process (in oppose to removal) can start as early as we'd like